### PR TITLE
Merge upstream from AnacondaRecipes

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
@conda-forge-admin , please rerender

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,9 +1,9 @@
 @echo on
-cd src\tools\msvc
+
+pushd src\tools\msvc
 
 echo $config-^>{openssl} = '%LIBRARY_PREFIX%'; >> config.pl
 echo $config-^>{zlib} = '%LIBRARY_PREFIX%';    >> config.pl
-echo $config-^>{python} = '%PREFIX%';          >> config.pl
```
Comment (sodre) [inline recipe/bld.bat]: Please keep the PL/Python support

File: recipe/bld.bat [hunk 2]
```diff
@@ -33,55 +33,8 @@ if "%ARCH%" == "32" (
...
-if errorlevel 1 exit 1
-call install.bat "%LIBRARY_PREFIX%"
-if errorlevel 1 exit 1
-
-echo Outputting versions: ctl, config, client
-pg_ctl --version
-pg_config --version
-psql --version
-
-REM On windows, it is necessary to start a server for the tests to connect to and run on
```
Comment (sodre) [inline recipe/bld.bat]: I think it is still valuable to run the tests during build time.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -22,44 +22,86 @@ requirements:
...
+        - perl                 # [win]
+      host:
+        # these are here for sake of run_exports taking effect
+        - krb5             # [not osx]
+        - openssl
+        - readline         # [not win]
+        - zlib
+        - {{ pin_subpackage('libpq', exact=True) }}
+      run:
+        - {{ pin_subpackage('libpq', exact=True) }}
```
Comment (isuruf) [inline recipe/meta.yaml]: You need `krb5, openssl, readline` here as these packages don't have run_exports in conda-forge

File: recipe/build.sh [hunk 4]
```diff
@@ -24,5 +20,4 @@ make check -C src/pl
 make check -C contrib
 make check -C src/interfaces/ecpg
 
-make install
 make install -C contrib
```
Comment (sodre) [inline recipe/build.sh]: You probably want to move `make install -C contrib` to install_db.sh as well.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -22,44 +22,86 @@ requirements:
     - {{ compiler('cxx') }}
     - {{ compiler('fortran') }}  # [not win]
   host:
-    - python
```
Comment (sodre) [inline recipe/meta.yaml]: Please don't drop the PL/Python dependency.

File: recipe/meta.yaml [hunk 6]
```diff
@@ -22,44 +22,86 @@ requirements:
...
+    - posix            # [win]
     - libxml2
-    - libxslt 1.1.*
-    - libuuid 1.0.*  # [not win]
+    - krb5             # [not osx]
+    - openssl
     - msinttypes  # [win and py27]
     - vc  # [win]
-  run:
-    - python
```
Comment (sodre) [inline recipe/meta.yaml]: Same here..

File: recipe/bld.bat (hunk 1 above)
Comment (mariusvniekerk) [reply recipe/bld.bat]: yeah this was mostly from the merge

File: recipe/meta.yaml (hunk 5 above)
Comment (mariusvniekerk) [reply recipe/meta.yaml]: Could we package the pl-python stuff seperately?  I don't know if we really want to force a python install for postgres

Comment (sodre) [reply recipe/meta.yaml]: As long as you make part of the main recipe that is okay with me. With conda-build-3, this should be a matter of adding a additional entry to `outputs`, right?

Comment (msarahan) [reply recipe/meta.yaml]: Ideally yes, but in practice, I'm not sure.  Depends on how much stuff needs to be done at compile time.  Please take a look and let me know if you have questions.

Comment (sodre) [reply recipe/meta.yaml]: @mariusvniekerk, I'm assuming you will take a look at it before merging.

Here are links to how pkgsrc does `postgresql10`

https://github.com/NetBSD/pkgsrc/tree/trunk/databases/postgresql10
https://github.com/NetBSD/pkgsrc/tree/trunk/databases/postgresql10-client
https://github.com/NetBSD/pkgsrc/tree/trunk/databases/postgresql10-server
https://github.com/NetBSD/pkgsrc/tree/trunk/databases/postgresql10-plpython

The PLIST files are the list of files that should be moved into each package.

Comment (msarahan) [reply recipe/meta.yaml]: If all it involves is splitting up files, it's definitely doable.  You can do that either with install scripts, or with explicit file lists (but not both at once).  https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#specifying-files-to-include-in-output

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 7]
```diff
@@ -22,44 +22,116 @@ requirements:
...
+        # these are here for sake of run_exports taking effect
+        - krb5     # [not osx]
+        - openssl
+        - readline         # [not win]
+        - zlib
+        - {{ pin_subpackage('postgresql', exact=True) }}
+        - python
+      run:
+        - {{ pin_subpackage('postgresql', exact=True) }}
+        - python
```
Comment (isuruf) [inline recipe/meta.yaml]: Same here. Need krb5, openssl, ...

File: recipe/meta.yaml [hunk 8]
```diff
@@ -22,44 +22,116 @@ requirements:
...
+      build:
+        # solely for sake of lining up vc versions and other runtimes
+        - {{ compiler('c') }}
+        - perl                 # [win]
+      host:
+        # these are here for sake of run_exports taking effect
+        - krb5     # [not osx]
+        - openssl
+        - readline         # [not win]
+        - zlib
```
Comment (isuruf) [inline recipe/meta.yaml]: Same here. Need krb5, openssl, ...

File: recipe/meta.yaml (hunk 7 above)
Comment (mariusvniekerk) [reply recipe/meta.yaml]: Does it not get it from the parent?

Comment (isuruf) [reply recipe/meta.yaml]: Ah, yes it does

**Comment by isuruf** [comment]
@mariusvniekerk, looks like the issue is that, the work directory is copied to python 3.6 build and for python 2.7 build, the work directory from 3.6 build is copied instead of original work directory. When the python 2.7 build is run, it doesn't re-compile the python sources, but tries to link them to python 2.7 which fails.

@msarahan, any ideas how to fix this?

**Comment by mariusvniekerk** [comment]
guess we can make clean in those dirs?

File: recipe/meta.yaml [hunk 9]
```diff
@@ -22,44 +22,135 @@ requirements:
...
+        - {{ pin_subpackage('libpq', exact=True) }}
+      run:
+        - {{ pin_subpackage('libpq', exact=True) }}
+        - krb5
+        - openssl
+        - readline         # [not win]
+        - zlib
+        - libxml2
```
Comment (isuruf) [inline recipe/meta.yaml]: Shouldn't this be in host as well?

**Comment by mariusvniekerk** [comment]
Ok windows builds seem to be blocked on krb5 stuff.  guess we need to merge that one first?

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Comment by ocefpaf** [comment]
I guess we need this one to make #30 work, right?

**Comment by mariusvniekerk** [comment]
Probably.

**Comment by mariusvniekerk** [comment]
@ocefpaf THIS PASSES!

**Comment by mariusvniekerk** [comment]
@jakirkham @msarahan So the thing thats a bit odd with this one is that whilst we generate a python matrix, we probably need to keep them together for the build, since only 1 of the outputs depends on python.

What is the correct approach to deal with this case?

**Comment by ocefpaf** [comment]
@sodre are you OK merging this or do you still want to perform some changes?

**Comment by ocefpaf** [comment]
@sodre we can address your comments in another PR. Merging this to unblock the bot :grimacing:

**Comment by sodre** [comment]
I'm good! great job @mariusvniekerk!

**Pull request merged**